provisioning/baremetal_pod.go

```diff
@@ -468,10 +475,54 @@ func NewMetal3Deployment(targetNamespace string, images *Images, config *metal3i
 	}
 }
 
-func GetExistingMetal3Deployment(client appsclientv1.DeploymentsGetter, targetNamespace string) (bool, error) {
-	existing, err := client.Deployments(targetNamespace).Get(context.Background(), baremetalDeploymentName, metav1.GetOptions{})
+func annotateExistingMetal3Deployment(existing *appsv1.Deployment) {
+	curAnnotations := existing.GetAnnotations()
+	if _, ok := curAnnotations[cboOwnedAnnotation]; ok {
+		// metal3 deployment already owned by CBO
+		return
+	}
+	cboAnnotations := map[string]string{}
+	cboAnnotations[cboOwnedAnnotation] = ""
+	modified := resourcemerge.BoolPtr(true)
+	resourcemerge.MergeMap(modified, &cboAnnotations, existing.GetAnnotations())
+	existing.SetAnnotations(cboAnnotations)
+	return
+}
+
+func ApplyNewOrUpdatedDeployment(client appsclientv1.DeploymentsGetter, targetNamespace string, metal3Deployment *appsv1.Deployment, generations []osoperatorv1.GenerationStatus, componentName string) error {
+	finalDeployment := metal3Deployment
+
+	// Check if there is an existing metal3 deployment
+	existing, err := client.Deployments(targetNamespace).Get(context.Background(), BaremetalDeploymentName, metav1.GetOptions{})
+
 	if existing != nil && err == nil {
-		return true, nil
+		if equality.Semantic.DeepEqual(metal3Deployment.Spec, existing.Spec) {
```

Comment:
Do we need the annotation to be handled as a separate case? Can we simplify this by building the structure for what we want the deployment to look like, including the annotation, and then always updating it? If there is no change, nothing will happen. If there are any changes, including the annotation, the deployment would be updated all at once.

Reply:
That is exactly the semantics ApplyNewOrUpdatedDeployment() is providing. 

But, within the function we need to look at different parts of the Deployment object and update them independently to prep the finalDeployment object. Then at https://github.com/openshift/cluster-baremetal-operator/pull/65/files#diff-1575ce96065be1a97bee923445ae60115c8ce02b4a2736788012df8162407100R508, this finalDeployment is applied.

Reply:
Why can't the Deployment be computed all at once, then applied via Create() if it does not exist or Update() if it does exist?

Reply:
I am looking into how to best use of some Create() vs Update() logic already built into resourceapply.applyDeployment().

Reply:
It's my understanding that ApplyDeployment() already implements what you're trying to do here.